### PR TITLE
fixes k3d local installation

### PR DESCRIPTION
**Pull request by UncleGedd**
- Fix: metal-lb's speaker pod was failing due to not being able to access ConfigMaps
- Fix: Promtail pod wouldn't start because it couldn't access `/etc/machine-id`
- Removes extraneous args from the `k3d` command
- Adds artifacts to k3d .gitignore

### todo
- make sure we are dogfooding `k3d/local`?

### CODE REVIEW

File: k3d/local/manifests/metallb/metallb-native.yaml [hunk 1]
```diff
@@ -1308,6 +1308,7 @@ rules:
   - ""
   resources:
   - secrets
+  - configmaps
```
Comment (mjnagel) [inline k3d/local/manifests/metallb/metallb-native.yaml:1311]: I'm curious on the error you saw here. This file comes straight from the upstream source I believe (https://raw.githubusercontent.com/metallb/metallb/v0.13.10/config/manifests/metallb-native.yaml) and also does not have configmaps upstream.

File: k3d/local/scripts/k3d.sh [hunk 2]
```diff
@@ -12,8 +12,9 @@ else
     --k3s-arg "--disable=traefik@server:*" \
     --k3s-arg "--disable=metrics-server@server:*" \
     --k3s-arg "--disable=servicelb@server:*" \
-    --port 443:443@loadbalancer \
-    --port 80:8080@loadbalancer \
+    --volume /tmp/dubbd:/var/lib/rancher/k3s/storage@all \
+    --volume /etc/machine-id:/etc/machine-id \
+    --volume /var/log/pods:/var/log/pods \
```
Comment (mjnagel) [inline k3d/local/scripts/k3d.sh:17]: Also curious on this one...I've deployed with k3d without mounting this extra volume and saw no issues with promtail. It does normally log a warning that certain paths are directories vs log files but should still tail the log files okay?

Comment (UncleGedd) [reply k3d/local/scripts/k3d.sh]: You're right, this was leftover from testing, removed with latest commit

**Review by mjnagel — APPROVED** [review]
LGTM! Thanks for finding and fixing these issues.

**Pull request merged**